ghc-lib-gen/src/Ghclibgen.hs

```diff
@@ -374,6 +374,29 @@ commonBuildDepends =
   , "hpc == 0.6.*"
   ]
 
+-- | This utility factored out to avoid repetion.
+libBinParserModules :: IO ([Cabal], [Cabal], [String])
+libBinParserModules = do
+    lib <- mapM readCabalFile cabalFileLibraries
+    bin <- (:[]) <$> readCabalFile cabalFileBinary
+    parserModules <- calcParserModules
+    return (lib, bin, parserModules)
+
+-- | Call this after cabal file generation. These files are generated
+-- from '.hsc' files in the source tree and we prefer to ship those in
+-- the sdists rather than these. If we don't remove them, some
+-- ambiguity results. Cabal and stack are OK with that but bazel gets
```

Comment:
Note that even for Cabal and Stack it makes sense to not ship them. hsc2hs is platform dependent no including the generated files isn’t going to work in general.